providers/zohocrm/errors.go

```diff
@@ -0,0 +1,45 @@
+package zohocrm
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+/*
+The normal standard response from ZohoCRM look like this:
+{
+	"data": [{...}, {...}],
+	"info": {...}
+}
+*/
+
+// responseHandler wraps the http.StatusNotModified response into http.StatusOK
+// and sends back an empty data field response.
+func responseHandler(resp *http.Response) (*http.Response, error) { //nolint:cyclop
+	// responseData represents data we will send back when the response status code 304.
+	responseData := map[string]any{
+		"data": []any{},
+		"info": map[string]any{
+			"page":         1,
+			"more_records": false,
+		},
+	}
+
+	// When the ZohoCRM API responds with 304 status code,
+	// this indicates there is no data modified since the modification time provided.
+	// We modify the response to return 200, And an empty data field response.
+	if resp.StatusCode == http.StatusNotModified {
```

Comment:
@RajatPawar @laurenzlong I don't think we need to handle this, do we? We can just return 304 with data `nil`. Then it will be taken care of by the caller, right?

Reply:
I agree, Read()'s return logic should take a nil response and convert it into an empty map

Reply:
Okay, Just for clarification. If we don't handle this then we will be returning an error to the caller when there is no new records to return(when we receive 304). Since we are assuming any non 200>=x<299 status are errors. won't this be an issue to the caller?

Reply:
Actually, your approach makes sense and good point on how we are handling the error. Yeah, wrapping with 200 ok and empty array is good.

Reply:
Okay, so after discussing with others in my team. This is not the correct way to handle it. We should handle 3XX error correctly for all connectors. But for the scope of this task, we can merge it as is. I will create a new task to handle 3XX errors in general, but it will not be your task to do. 

TLDR, this PR is good for now.